promiscuous.js

```diff
@@ -91,7 +97,13 @@
           transform.call(value, resolve, reject);
       }
       catch (error) { reject(error); }
-    }, 1);
+    }
+    // Don't polyfill setImmediate, but use it if it's available.
+    if (window.setImmediate) {
+      window.setImmediate(fn);
+    } else {
+      setTimeout(fn, 0);
```

Comment:
ah, interesting twist — I didn’t realize promiscuous fell back on setTimeout so quickly in the first place. Typically a P polyfill uses the messaging API (polyfilling setImmediate, effectively), leaving setTimeout as a last ditch "it will probably still work, usually" solution for IE9 and below. That said, browsers without native Promise that aren’t IE 10 or 11 are now unusual enough that perhaps it no longer matters in 2017.